From 42361d423b32eb3e1dfa3c90275d415dfc455a4d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Tue, 8 Jul 2025 17:15:10 +0200
Subject: [PATCH 913/954] fix(st-uart): aarch32: fix ISR bit use for console
 flush
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use the transfer complete bit instead of the transmit data register
emptiness bit.

Fixes: 63af05e633ea ("fix(st-uart): aarch32: wait for UART ISR register TC bit for console flush")

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 0996ab2da16f886c9bc0e6646f24dd92e88af3a1 fix(st-uart): aarch32: fix ISR bit use for console flush]
Change-Id: I516dc0124fa238b2ed384ff31ea95e16faf7e63e
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/473406
Tested-by: Valentin CARON <valentin.caron@foss.st.com>
Reviewed-by: Valentin CARON <valentin.caron@foss.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 drivers/st/uart/aarch32/stm32_console.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/st/uart/aarch32/stm32_console.S b/drivers/st/uart/aarch32/stm32_console.S
index 1942c6bfb8..1a51350c10 100644
--- a/drivers/st/uart/aarch32/stm32_console.S
+++ b/drivers/st/uart/aarch32/stm32_console.S
@@ -234,7 +234,7 @@ func console_stm32_core_flush
 	/* Check Transmission Complete flag */
 tc_loop:
 	ldr	r1, [r0, #USART_ISR]
-	tst	r1, #USART_ISR_TXE
+	tst	r1, #USART_ISR_TC
 	beq	tc_loop
 1:
 	bx	lr
-- 
2.53.0

